src/TokenBuyer.sol

```diff
@@ -337,14 +340,20 @@ contract TokenBuyer is Ownable, Pausable, ReentrancyGuard {
     }
 
     function setPriceFeed(IPriceFeed newPriceFeed) external onlyOwner {
+        emit PriceFeedSet(address(priceFeed), address(newPriceFeed));
+
         priceFeed = newPriceFeed;
     }
 
     function setPayer(address newPayer) external onlyOwner {
+        emit PayerSet(payer, newPayer);
+
         payer = newPayer;
     }
 
     function setAdmin(address newAdmin) external onlyOwner {
```

Comment:
unrelated to this change, but related to this function:
perhaps we want to allow `admin` to be able to call `setAdmin`, wdyt?